dynamo/tools/clustering.py

```diff
@@ -545,16 +518,18 @@ def cluster_community(
                     X_data = adata.obsm[obsm_key]
                     neighbors(adata, X_data=X_data, result_prefix=obsm_key)
             else:
-                X_data = adata[:, adata.var.use_for_pca].layers[layer]
+                X_data = adata.layers[layer]
```

Comment:
use for pca is needed because dynamo keeps all genes while we often use top 2k genes for downstream analysis and a lot of genes may be set to be np.nan

Reply:
You are right. I was thinking about keeping consistency and let user to decide what columns to use outside this function because user may use use_for_umap and use_for_something or other keys. I will add a logging info here to know we use pca genes here.

Reply:
that is great, thanks! for any code changes, you may be a little more cautious because it may break up the current behaviors

Reply:
there is no such thing of use for umap, etc. in dynamo, we have use_pca, use_for_dynamics, use_for_transition, each corresponds to genes used in pca dimension reduction, and learning the kinetic parameters and velocity vectors and then the genes used in velocity projection to low dimensional space

Reply:
umap is generally done by taking the top 30 PCs for dimension reduction